examples/bri-3/src/bri/auth/capabilities/generate-nonce/generate.nonce.command.handler.ts

```diff
@@ -15,12 +13,7 @@ export class GenerateNonceCommandHandler
       command.publicKey,
     );
 
-    if (!bpiSubject.loginNonce || bpiSubject.loginNonce === '') {
-      bpiSubject.loginNonce = v4();
-      await this.authAgent.updateLoginNonce(bpiSubject);
-      return bpiSubject.loginNonce;
-    } else {
-      throw new Error(errorMessage.RELOGIN_FAILED);
```

Comment:
i changed this so new nonce is simply generated for every login

Reply:
not sure if we shall allow a relogin attempt if a user is already authenticated. Could you please describe this change?

Reply:
its just how every auth api i ever saw works, you want to login, you get jwt token, and also there was a situation during development where i was locked out of the app because of this, i could not make it work - i can try and reproduce this if you want, but can not remember exact steps now

the problem is you are using this to check if user is authenticated even though there is more to this then simply checking for presence of a nonce in db (for example, what if jwt expires, how will you know to allow re-login?)

what is the situation to not allow re-login?

Reply:
how was this supposed to work, i just opened the app to reproduce bug i mentioned and i can not login
so frontend wants to generate nonce, but if nonce is already in db, this line:
```
 if (!bpiSubject.loginNonce || bpiSubject.loginNonce === '') {
 }
 else throw RE-LOGIN
```

will always fail and there is no way to recover from this error...
i assume both 2 following would not work:
- if token expires, nonce is still there but token doesnt work anymore, but you can not relogin
- if  you accidentaly dont save your jwt (eg. you want to test in postman) you need to be able to relogin, this way it is literally impossible

Reply:
Okay, so basically the removal of this code snippet protects us with all relogin attempts?

Reply:
not sure about relogin attempts, basically previous code was stopping re-generation of nonce, but you could still relogin if you wanted
the problem with previous approach was if nonce is generated but login not finished (which can happen if user for example doesnt sign nonce) then you are stuck without a way to recover because nonce generation and login are separated coupled endpoints in that case

frontend -> nonce generation endpoint -> sign nonce -> login endpoint

if sign nonce is skipped, nonce is not removed in db (login endpoint removes it) and there is no way to recover